cmd/gnoblog-cli/post.go

```diff
@@ -0,0 +1,206 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"github.com/gnolang/gno/tm2/pkg/commands"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/gnolang/gno/gno.land/pkg/gnoclient"
+	"github.com/gnolang/gno/gnovm/pkg/gnoenv"
+	"github.com/peterbourgon/ff/v3/ffcli"
+)
+
+type cliCfg struct {
+	Publish       bool
+	Edit          bool
+	GasWanted     int64
+	GasFee        string
+	ChainId       string
+	BlogRealmPath string
+
+	KeyName               string
+	GnoHome               string
+	Remote                string
+	Quiet                 bool
+	InsecurePasswordStdIn bool
+}
+
+func newPostCommand(io commands.IO) *ffcli.Command {
+	var (
+		fs  = flag.NewFlagSet("post", flag.ExitOnError)
+		cfg = &cliCfg{}
+	)
+
+	// Register flagset
+	cfg.registerFlags(fs)
+
+	// Make Post command
+	return &ffcli.Command{
+		Name:       "post",
+		ShortUsage: "post <FILE OR FILES_DIR> [flags]",
+		LongHelp:   `Post one or more files. Passing in a file will post that single file, while passing in a directory will search for all .md files and batch post them.`,
+		FlagSet:    fs,
+		Exec: func(_ context.Context, args []string) error {
+			return cfg.execPost(io, args)
+		},
+	}
+}
+
+// Register flags
+func (cfg *cliCfg) registerFlags(fs *flag.FlagSet) {
+	fs.StringVar(&cfg.KeyName,
+		"key",
+		"",
+		"name of keypair to use for deployment",
+	)
+	fs.BoolVar(&cfg.Publish,
+		"publish",
+		false,
+		"publish blogpost",
+	)
+	fs.BoolVar(&cfg.Edit,
+		"edit",
+		false,
+		"edit mode",
+	)
+	fs.Int64Var(&cfg.GasWanted,
+		"gas-wanted",
+		5000000,
+		"gas requested for tx",
+	)
+	fs.StringVar(&cfg.GasFee,
+		"gas-fee",
+		"1000000ugnot",
+		"gas payment fee",
+	)
+	fs.StringVar(&cfg.ChainId,
+		"chainid",
+		"dev",
+		"chain ID",
+	)
+	fs.StringVar(&cfg.BlogRealmPath,
+		"pkgpath",
+		"gno.land/r/gnoland/blog",
+		"blog realm path",
+	)
+	fs.StringVar(&cfg.GnoHome,
+		"home",
+		gnoenv.HomeDir(),
+		"home directory",
+	)
+	fs.StringVar(&cfg.Remote,
+		"remote",
+		"localhost:26657",
+		"remote node URL",
+	)
+	fs.BoolVar(&cfg.InsecurePasswordStdIn,
+		"insecure-password-stdin",
+		false,
+		"WARNING! take password from stdin",
+	)
+}
+
+func (cfg *cliCfg) execPost(io commands.IO, args []string) error {
+	if len(args) > 1 {
+		return errInvalidNumberOfArgs
+	}
+
+	// Stat passed in arg
+	fileInfo, err := os.Stat(args[0])
+	if err != nil {
+		return errInvalidPath
+	}
```

Comment:
adding more context to the error can be helpful
```suggestion
	// Stat passed in arg
	fileInfo, err := os.Stat(args[0])
	if err != nil {
		return fmt.Errorf("unable to stat %q: %w", args[0], err)
	}
```

Reply:
[6f5d2b6](https://github.com/gnolang/blog/pull/63/commits/6f5d2b60ac413d48cc00a45a2c618b83ebb8002f)